esda/getisord.py

```diff
@@ -530,22 +552,115 @@ def by_col(
             outvals=outvals,
             stat=cls,
             swapname=cls.__name__.lower(),
-            **stat_kws
+            **stat_kws,
         )
 
 
+def _infer_star_and_structure_w(weights, star, transform):
+    assert transform.lower() in ("r", "b"), (
+        f'Transforms must be binary "b" or row-standardized "r".'
+        f"Recieved: {transform.upper()}"
+    )
+    adj_matrix = weights.sparse
+    diagonal = adj_matrix.diagonal()
+    zero_diagonal = (diagonal == 0).all()
+
+    # Gi has a zero diagonal, Gi* has a nonzero diagonal
+    star = (not zero_diagonal) if star is None else star
+
+    # Want zero diagonal but do not have it
+    if (not zero_diagonal) & (star is False):
```

Comment:
It would, but this is where the the trickiness of this API comes in. There are *three* correct values for `star`: `True`/`False`/`float`. If `star` is a float, we need to use it as the default value for the self-weight, but if it's `True`, we need to set a default.

Reply:
Ah. Def something that I missed.